File: split_example.py

import jpeg_spliter
path = "unity_texture2d_big.jpg"
write_to_file = False
test_time = 200
f = open(path, "rb")
data = bytearray(f.read())
f.close()

param = jpeg_spliter.SpliterParam(
    singleWidth=1280,
    singleHeight=768,
    col=4,
    row=5,
    dri=8
)

cost = 0

for i in range(test_time):
    icost, out = jpeg_spliter.split(data, param)
    cost += icost
    if write_to_file:
        for i in range(len(out)):
            f = open(f"sub/{i}.jpg", "wb")
            f.write(out[i])
            f.close()
print(f"avg {test_time} cost:{cost/test_time}, fps: {test_time/cost}")

File: CSharp/SplitExample.cs
SpliterParam param = new SpliterParam
{
    SingleHeight = 768,
    SingleWidth = 1280,
    Col = 4,
    Row = 5,
    Dri = 8,
    Trim = false
};
string relativePath = "../../../../";
byte[] data = File.ReadAllBytes(relativePath+"unity_texture2d_big.jpg");
int length = data.Length;
List<byte>[] output = new List<byte>[param.Col*param.Row];
int testTime = 200;
double cost = 0;
for (int i = 0; i < testTime; i++)
{
    cost += Spliter.Split(data, length, param, output);
}
for(int i = 0; i < output.Length; i++)
{
    File.WriteAllBytes($"{relativePath}sub/{i}.jpg", output[i].ToArray());
}
Console.WriteLine($"avg {testTime} cost: {cost/testTime}, fps:{testTime/cost}");